### ISSUE TITLE
`Server.acceptable` not merging with `options.formatters` for constructor the Server

### ISSUE THREAD

**Issue by KostyaTretyak** [restify/node-restify#1280]
```js
const restify = require('restify');

const formatters =
{
  'application/json; q=0.2': () =>{},
  'text/plain; q=0.4': () =>{},
  'application/octet-stream; q=0.1': () =>{}
};

const server = restify.createServer({formatters});

console.log( server.acceptable );
```

Returns:

```txt
[ 'application/json',
  'text/plain',
  'text/plain',
  'application/octet-stream',
  'application/json',
  'application/javascript',
  'application/octet-stream' ]
```

**Comment by KostyaTretyak** [comment]
We can rewrite [this](https://github.com/restify/node-restify/blob/ed43e870f589def25ae4a3452d5d43d472e0cdc7/lib/server.js#L145-L149):

```js
arr = arr
.sort( (a, b) => b.q - a.q )
.map( a => a.t )
.filter( (item, pos, self) =>
{
  return self.indexOf(item) == pos;
});
```

**Comment by retrohacker** [comment]
Hey @KostyaTretyak,

Thank you for taking the time to open this, and for providing a repro case! :heart:

It looks like this is a duplicate of #1219

Closing for now, please correct me if I'm missing a nuance.

**Comment by retrohacker** [comment]
P.S. there is a closed PR that contains most of the work necessary to fix this issue (linked to in #1219), I believe @anup4f82 is having a similar problem. If either of you can poke around that PR and isolate the changes necessary to fix your problem, I'll be happy to do a code review for you and we can get this merged into the codebase :smile:
